data-prepper-plugins/sns-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/SNSSinkServiceTest.java

```diff
@@ -0,0 +1,139 @@
+        this.numberOfRecordsSuccessCounter = mock(Counter.class);
+        this.numberOfRecordsFailedCounter = mock(Counter.class);
+        this.publishBatchResponse = mock(PublishBatchResponse.class);
+        this.sdkHttpResponse = mock(SdkHttpResponse.class);
+
+        when(pluginMetrics.counter(SNSSinkService.NUMBER_OF_RECORDS_FLUSHED_TO_SNS_SUCCESS)).thenReturn(numberOfRecordsSuccessCounter);
+        when(pluginMetrics.counter(SNSSinkService.NUMBER_OF_RECORDS_FLUSHED_TO_SNS_FAILED)).thenReturn(numberOfRecordsFailedCounter);
+        when(snsClient.publishBatch(any(PublishBatchRequest.class))).thenReturn(publishBatchResponse);
+        when(publishBatchResponse.sdkHttpResponse()).thenReturn(sdkHttpResponse);
+        when(publishBatchResponse.sdkHttpResponse().statusCode()).thenReturn(new Random().nextInt());
+        when(sdkHttpResponse.statusCode()).thenReturn(200);
+
+    }
+
+
+    private SNSSinkService createObjectUnderTest(){
+        return new SNSSinkService(snsSinkConfig,
+                snsClient,
+                pluginMetrics, mock(PluginFactory.class), mock(PluginSetting.class));
+    }
+
+    @Test
+    public void sns_sink_test_with_empty_collection_records(){
+        numberOfRecordsSuccessCounter = mock(Counter.class);
+        SNSSinkService snsSinkService = createObjectUnderTest();
+        snsSinkService.output(List.of());
+        verifyNoInteractions(numberOfRecordsSuccessCounter);
+        verifyNoInteractions(numberOfRecordsFailedCounter);
+    }
+
+    @Test
+    public void sns_sink_test_with_single_collection_record_success_push_to_sns(){
+        when(sdkHttpResponse.statusCode()).thenReturn(500);
+        SNSSinkService snsSinkService = createObjectUnderTest();
+        final Record<Event> eventRecord = new Record<>(JacksonEvent.fromMessage("{\"message\":\"c3f847eb-333a-49c3-a4cd-54715ad1b58a\"}"));
+        Collection<Record<Event>> records = List.of(eventRecord);
+        snsSinkService.output(records);
+        verify(numberOfRecordsSuccessCounter).increment(records.size());
+    }
+
+    @Test
+    public void sns_sink_test_with_single_collection_record_failed_to_push_to_sns(){
+        final Counter abc = mock(Counter.class);
+        when(pluginMetrics.counter(SNSSinkService.NUMBER_OF_RECORDS_FLUSHED_TO_SNS_FAILED)).thenReturn(abc);
+        final SnsException snsException = (SnsException)SnsException.builder().message("internal server error").awsErrorDetails(AwsErrorDetails.builder().errorMessage("internal server error").build()).build();
+        when(snsClient.publishBatch(any(PublishBatchRequest.class))).thenThrow(snsException);
+        SNSSinkService snsSinkService = createObjectUnderTest();
+        final Record<Event> eventRecord = new Record<>(JacksonEvent.fromMessage("{\"message\":\"c3f847eb-333a-49c3-a4cd-54715ad1b58a\"}"));
+        Collection<Record<Event>> records = List.of(eventRecord);
+        snsSinkService.output(records);
+        verify(abc).increment(records.size());
```

Comment:
This should also verify that the SNS client was actually called. Also, it should verify that the failure is handled correctly, say by pushing to the DLQ.

Reply:
Thanks @dlvenable for your feedback, Sure, it will available in next commit for review.